src/assemble_workflow/fpm_builder.py

```diff
@@ -0,0 +1,94 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import os
+import shutil
+import subprocess
+from assemble_workflow.bundle_recorder import BundleRecorder
+
+
+class FpmBuilder:
+    def build(self, bundle_recorder: BundleRecorder, archive_path: str, distribution: str) -> None:
+        manifest_data = bundle_recorder.bundle_manifest.data["build"]
+        product_name = product_name_alt = manifest_data["name"].lower()
+        if product_name == "opensearch dashboards":
+            product_name = "-".join(product_name.split())
+            product_name_alt = "_".join(product_name.split("-"))
+        root = os.path.abspath(os.path.join(__file__, os.pardir, os.pardir, os.pardir))
+        scripts_path = os.path.join(root, "scripts", "pkg")
+        architecture_alt = "x86_64" if manifest_data["architecture"] == "x64" else "aarch64"
```

Comment:
Everywhere you look at `manifest_data["..."]` you should be accessing properties of the manifest, e.g. `manifest.build.architecture`. 

Every time you do an `if` on a well defined list such as `x64` vs. `arm64`, create a property inside `manifest.build` to return it, and use a table like for EXTENSIONS so that we get an error if the value is unexpected. Call the method something like `manifest.build.alt_architecture` that makes general sense and add tests for it.

Reply:
Resolved both:
1. Using manifest.build.<>.
2. Setup a map / dictionary to map `distribution_architecture` to corresponding values.

Thanks.